internal/parser/parser.go

```diff
@@ -21,6 +26,809 @@ type Scanner interface {
 }
 
 // NewAST parses Flux query and produces an ast.Program.
-func NewAST(src Scanner) (*ast.Program, error) {
-	return nil, errors.New("implement me")
+func NewAST(src Scanner) *ast.Program {
+	p := &parser{
+		s: &scannerSkipComments{
+			Scanner: src,
+		},
+	}
+	return p.program()
+}
+
+// scannerSkipComments is a temporary Scanner used for stripping comments
+// from the input stream. We want to attach comments to nodes within the
+// AST, but first we want to have feature parity with the old parser so
+// the easiest method is just to strip comments at the moment.
+type scannerSkipComments struct {
+	Scanner
+}
+
+func (s *scannerSkipComments) Scan() (pos token.Pos, tok token.Token, lit string) {
+	for {
+		pos, tok, lit = s.Scanner.Scan()
+		if tok != token.COMMENT {
+			return pos, tok, lit
+		}
+	}
+}
+
+func (s *scannerSkipComments) ScanWithRegex() (pos token.Pos, tok token.Token, lit string) {
+	for {
+		pos, tok, lit = s.Scanner.ScanWithRegex()
+		if tok != token.COMMENT {
+			return pos, tok, lit
+		}
+	}
+}
+
+type parser struct {
+	s        Scanner
+	pos      token.Pos
+	tok      token.Token
+	lit      string
+	buffered bool
+}
+
+func (p *parser) program() *ast.Program {
+	program := &ast.Program{}
+	program.Body = p.statementList(token.EOF)
+	return program
+}
+
+func (p *parser) statementList(eof token.Token) []ast.Statement {
+	var stmts []ast.Statement
+	for {
+		if _, tok, _ := p.peek(); tok == eof || tok == token.EOF {
+			return stmts
+		}
+		stmts = append(stmts, p.statement())
+	}
+}
+
+func (p *parser) statement() ast.Statement {
+	switch _, tok, lit := p.peek(); tok {
+	case token.IDENT:
+		if lit == "option" {
+			return p.optionStatement()
+		}
+		return p.identStatement()
+	case token.RETURN:
+		return p.returnStatement()
+	case token.INT, token.FLOAT, token.STRING, token.DIV,
+		token.DURATION, token.PIPE_RECEIVE, token.LPAREN, token.LBRACK, token.LBRACE,
+		token.ADD, token.SUB, token.NOT:
+		return p.expressionStatement()
+	default:
+		// todo(jsternberg): error handling.
+		p.consume()
+		return nil
+	}
+}
+
+func (p *parser) optionStatement() ast.Statement {
+	p.expect(token.IDENT)
+	return p.optionDeclaration()
+}
+
+func (p *parser) optionDeclaration() ast.Statement {
+	switch _, tok, _ := p.peek(); tok {
+	case token.IDENT:
+		decl := p.variableDeclaration()
+		return &ast.OptionStatement{
+			Declaration: decl,
+		}
+	case token.ASSIGN:
+		expr := p.assignStatement()
+		return &ast.VariableDeclaration{
+			Declarations: []*ast.VariableDeclarator{{
+				ID:   &ast.Identifier{Name: "option"},
+				Init: expr,
+			}},
+		}
+	default:
+		// todo
+		p.consume()
+		return nil
+	}
+}
```

Comment:
Where is the ExpressionSuffix branch in this rule?

Reply:
No tests actually utilize this branch so I should add one. This was a placeholder while I was refactoring everything. I'll fix it now.